iree/hal/cuda/status_util.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_HAL_CUDA_STATUS_UTIL_H_
+#define IREE_HAL_CUDA_STATUS_UTIL_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/cuda/cuda_headers.h"
+
+namespace iree {
+namespace hal {
+namespace cuda {
+struct DynamicSymbols;
+}  // namespace cuda
+}  // namespace hal
+}  // namespace iree
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Converts a CUresult to an iree_status_t.
+//
+// Usage:
+//   iree_status_t status = CU_RESULT_TO_STATUS(cuDoThing(...));
+#define CU_RESULT_TO_STATUS(syms, expr, ...) \
+  iree_hal_cuda_result_to_status(syms, (syms->expr), __FILE__, __LINE__)
+
+// IREE_RETURN_IF_ERROR but implicitly converts the CUresult return value to
+// a Status.
+//
+// Usage:
+//   CUDA_RETURN_IF_ERROR(cuDoThing(...), "message");
+#define CUDA_RETURN_IF_ERROR(syms, expr, ...)                               \
+  IREE_RETURN_IF_ERROR(                                                     \
+      iree_hal_cuda_result_to_status(syms, syms->expr, __FILE__, __LINE__), \
```

Comment:
You probably want matching ()'s around the syms->expr here and below, as well as around syms itself. clang-tidy has hints for this, if you're using that. With vscode's clang-tidy extension I get little squigglies which is a nice reminder :)